### DISCUSSION TITLE
[IntelliSense] Support render props

### DISCUSSION THREAD

**Discussion by kripod** [tailwindlabs/tailwindcss#9566] [Ideas]
Sometimes, developers may want to provide a [render prop](https://beta.reactjs.org/apis/react/Children#calling-a-render-prop-to-customize-rendering) for customizing the underlying semantics of their component, e.g.:

```tsx
import { clsx } from "clsx";
import * as React from "react";

type MyButtonProps = React.ComponentPropsWithRef<"button"> & {
  render?: (
    props: React.ComponentPropsWithoutRef<"button"> &
      Required<
        Pick<
          React.ComponentPropsWithoutRef<"button">,
          "disabled" | "className" | "children"
        >
      >,
  ) => React.ReactNode;
};

function MyButton({
  disabled = false,
  className,
  render,
  children,
  ...restProps
}: MyButtonProps) {
  return (
    <>
      {(render ?? ((props) => <button type="button" {...props} />))({
        disabled,
        className: clsx(
          "transition focus:outline-none focus-visible:ring",
          disabled && "cursor-not-allowed opacity-40",
          className,
        ),
        children: children ?? null,
        ...restProps,
      })}
    </>
  );
}

function MyApp() {
  return (
    <MyButton
      render={({ className, children }) => (
        <a href="#_" className={className}>
          {children}
        </a>
      )}
    >
      Click me
    </MyButton>
  );
}
```

For those cases, simple objects with a `className` property should be supported as autocompletion and linting targets.

Anything similar to:

```ts
{
  className: "transition focus:outline-none focus-visible:ring",
}
```

or even:

```ts
{
  className: clsx("transition focus:outline-none focus-visible:ring"),
}
```

should be detected as a list of class names.

**Comment by wongjn** [comment]
You can try using the `tailwindCSS.experimental.classRegex` configuration option. There's not much "official" documentation on it but an internet search should give some hints on its use.

**Comment by kripod** [comment]
Whoa, that’s awesome, I didn’t know about that. Found the docs right here: https://github.com/tailwindlabs/tailwindcss/issues/7553#issuecomment-735915659

This seems to do the trick for my case:

```jsonc
// .vscode/settings.json
{
  "tailwindCSS.experimental.classRegex": [["clsx\\(([^)]*)\\)", "\"([^\"]*)\""]]
}
```

**Comment by kripod** [comment]
I’ve just opened a PR solving a similar issue within the Prettier-based class name auto-sorter plugin: https://github.com/tailwindlabs/prettier-plugin-tailwindcss/pull/97
